torch/csrc/distributed/c10d/reducer.h

```diff
@@ -115,6 +115,11 @@ class Reducer {
     std::vector<size_t> offsets;
     std::vector<size_t> lengths;
 
+    // Per-bucket tensor for indicating if variable is used locally during the
+    // current iteration. This tensor will be allreduced for figuring out the
+    // globally unused parameters. 1 means locally used and 0 locally unused.
+    at::Tensor local_used;
```

Comment:
As it seems it's not easy to coalesce the bitmap into the existing bucket allreduce, I wonder whether we should make this per-backward bitmap as @pietern suggested, so that we don't have to introduce multiple additional allreduce.

This allreduce can even be done at the end of the forward pass in `prepare_for_backward` where we find all unused params. In that way we will have minimum perf impact.

Reply:
OK let's keep using integer tensor then. int32 should be good enough? as the number of ranks cannot be that big.